Library/ViewModels/PledgeViewModel.swift

```diff
@@ -502,15 +502,25 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
       scaFlowCompletedWithSuccess.combineLatest(with: creatingContext).ignoreValues()
     )
 
+    let checkoutId = createBackingEvents.values()
+      .map { $0.checkoutId }
+
     let thanksPageData = createBackingDataAndIsApplePay
-      .map { data, isApplePay -> ThanksPageData in
-        let checkoutPropsData = checkoutPropertiesData(from: data, isApplePay: isApplePay)
+      .combineLatest(with: checkoutId)
+      .map(unpack)
+      .map { data, isApplePay, checkoutId -> ThanksPageData in
+        let checkoutPropsData = checkoutPropertiesData(
+          from: data,
+          checkoutId: checkoutId,
+          isApplePay: isApplePay
+        )
 
         return (data.project, data.reward, checkoutPropsData)
       }
 
-    self.goToThanks = thanksPageData
-      .takeWhen(createBackingCompletionEvents)
+    self.goToThanks = createBackingCompletionEvents
```

Comment:
Why was this change needed? From what I can see the way it was before should still work, and is one line of code shorter 😄

Reply:
I forgot to mention here... using `takeWhen` broke some other test and this solution was the one that made all of them pass 😅

Reply:
I think we might want to be pretty specific that this should only fire when the `createBackingCompletionEvents` emit 🤔 Do you know what broke/why?

Reply:
I noticed that in some tests, `thanksPageData` emits before `createBackingCompletionEvents` and in others, like the one I've mentioned, the order changes and the test fails. Using `combineLatest` we don't depend on the order that the signals emit and it avoids a race condition

Reply:
I think it's just that `combineLatest` will also emit regardless of the order, in this case we just want to be sure that we only `goToThanks` when `createBackingCompletionEvents` emits. Given that this is tricky to do regression testing on I would try to avoid this change. Let's rather dig into _why_ the tests break with your change? I can take a look with you too.

Reply:
Ok, I see what's happening - previously the `thanksPageData` had all of the necessary data before the completion events emit. This changes that because now we want the `checkoutId` in the `ThanksPageData` which causes a bit of a race condition. I would move the creation of the `thanksPageData` to be a a signal that is `takePairWhen` `createBackingCompletionEvents` and retrieve the `checkoutId` right at the end.

Reply:
^ +1

Reply:
Another option would be using `zip` instead of `combineLatest`. This guarantees that `goToThanks` only emits after both signals emit. But just for curiosity, what are the disadvantages of using `.combineLatest` in this case?

Reply:
I think the concern is that `goToThanks` could fire before `createBackingCompletionEvents` during a retry flow, causing us to send the user to thanks page before we've confirmed that we're clear to do so, particularly during the SCA flow. We might be prematurely sending users to the thanks page before they've completed the SCA auth flow.

Reply:
> what are the disadvantages of using .combineLatest in this case?

I think the main concern is just that it doesn't accurately represent what we want to happen. Previously this was a `takeWhen` because that accurately describes the event, "take this data when this other thing happens". This has changed because we now require some data from the thing that has happened, but `takePairWhen` accurately describes that - "take some data from this other event, combine it with what we have, but only when that other thing happens". `combineLatest` on the other hand blurs this by saying "combine all this data when either of these things happen".